addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/ComponentFan.java

```diff
@@ -45,12 +41,11 @@
         protected String payload_off = "OFF";
     };
 
-    public ComponentFan(ThingUID thing, HaID haID, String configJSON,
-            @Nullable ChannelStateUpdateListener updateListener, Gson gson) {
-        super(thing, haID, configJSON, gson, Config.class);
+    public ComponentFan(CFactory.ComponentConfiguration builder) {
```

Comment:
Could you rename "builder" to "cconfig" or even "componentConfig" in all components? That'd be sweet.